src/main/java/com/reverse/userservice/controllers/UserController.java

```diff
@@ -100,4 +103,36 @@ public ResponseEntity editUser(@RequestBody UserEdit userEdit) {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
         }
     }
+
+    /**
+     * @Author: Alberto Figallo
+     * */
+    @PatchMapping(path = "/updateUser", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity updateUser(@RequestBody User user) {
+        log.debug("Begin updateUser");
+        List<String> availableFields = Arrays.asList("username","email","firstName","lastName","password","dateOfBirth","gender","branch","profilePicture");
+        User userData = userService.getUserByID(user.getId());
+
+        if (userData != null) {
+            User updUser = userData;
+            for (String field : availableFields) {
+                switch (field){
+                    case "username": if (user.getUsername() != null) updUser.setUsername(user.getUsername()); break;
+                    case "email": if (user.getEmail() != null) updUser.setEmail(user.getEmail()); break;
+                    case "firstName": if (user.getFirstName() != null) updUser.setFirstName(user.getFirstName()); break;
+                    case "lastName": if (user.getLastName() != null) updUser.setLastName(user.getLastName()); break;
+                    case "password": if (user.getPassword() != null) updUser.setPassword(user.getPassword()); break;
+                    case "dateOfBirth": if (user.getDateOfBirth() != null) updUser.setDateOfBirth(user.getDateOfBirth()); break;
+                    case "gender": if (user.getGender() != null) updUser.setGender(user.getGender()); break;
+                    case "branch": if (user.getBranch() != null) user.setBranch(user.getBranch()); break;
+                    case "profilePicture": if (user.getProfilePicture() != null) user.setProfilePicture(user.getProfilePicture()); break;
```

Comment:
I've got fixed tests added, but it looks like there is a mistake in the code here. I think these are supposed to be setting the Branch and profilePicture of the updUser.
```suggestion
                    case "branch": if (user.getBranch() != null) updUser.setBranch(user.getBranch()); break;
                    case "profilePicture": if (user.getProfilePicture() != null) updUser.setProfilePicture(user.getProfilePicture()); break;
```